### PR TITLE
feat(ideas cli): wire strategy-backed proposers into propose, cycle, and replay (#1164 stage 2)

### PR DESCRIPTION
**Pull request by Solders-Girdles**
## Summary

Stage 2 of #1164 (strategy→proposer convergence): the `ideas` CLI becomes the composition root that runs live-trade strategies as snapshot proposers, completing the operator surface over the Stage-1 `SnapshotStrategyProposer` (#1165).

- **`ideas propose-strategy --strategy {baseline-spot,baseline-perps}`** — runs a live-trade strategy over a local `MarketSnapshot` fixture through the audited `TradeIdeaService`, with the budget-backed sizing bridge (`_LazyBudgetSizingBridge`) injected at the composition root: sizing denominates against operator-attested equity exactly like the approval gate, and no-signal runs stay read-only (budget is never read or seeded).
- **`ideas cycle --proposer strategy-baseline-spot|strategy-baseline-perps`** — strategy-backed proposers are selectable but **opt-in**: the default proposer set stays `(baseline, regime-aware)` until replay parity is demonstrated, so scheduled turns (`scripts/ops/stage1_cycle_turn.sh`, `CYCLE_PROPOSERS`) are behaviorally unchanged.
- **`ideas replay strategy`** — scores a strategy over candle history; the min-history floor is derived from the live configuration (`max(long-MA, RSI+1) + crossover lookback = 23`) rather than CLI-tunable windows, because the parity lane replays the live configuration as-is. **`ideas replay tournament`** accepts strategy-backed ids beside `baseline-ma-<short>-<long>` for head-to-head parity/calibration evidence.
- **`--price-precision`** surfaced on `propose-strategy`, `cycle`, and `replay strategy`: sub-cent symbols opt into finer quantization instead of failing closed (Stage-1 fail-closed validation untouched; on `cycle` the knob applies uniformly to all proposers this turn, default `0.01` preserves behavior).
- **No new import edges**: strategies (`SpotStrategy`/`BaselinePerpsStrategy`) are constructed only in the CLI through the structural `SnapshotDecider` seam; `strategy_tools` still has no `live_trade` import. Boundary guard green.
- **Runbook** (`docs/paper_trading.md`): scheduled-turn section documents the `ideas budget set --account-equity` attestation prerequisite, the opt-in strategy proposers, and the sub-cent precision knob; the honest-day procedure gains the `propose-strategy` alternative.

## Verification

- End-to-end offline: `propose-strategy` → `approve` (real policy gate over attested budget) → `execute-paper` fill → `audit verify` intact; `cycle --proposer strategy-baseline-spot` rerun over the same snapshot dedupes on the identical `decision_id`; `replay strategy` scores the proposer over fixture history.
- `uv run pytest tests/unit -n auto -q` — 6368 passed.
- `make ci-required` — green (lint, mypy, unit, import boundaries, docs audits, stage1 rails smoke).
- New CLI tests: `test_ideas_propose_strategy.py` (executable sizing, attested-equity denomination, lazy budget read, determinism/dup refusal, sub-cent fail-closed + `--price-precision` unlock), `test_ideas_replay_strategy.py` (report envelope, min-history floor, noop), plus cycle opt-in selection and mixed tournament coverage.

Remaining on #1164 after this: Stage 3 — stateful strategies (mean-reversion cooldown externalized, regime-switcher via fresh detector per snapshot; ensemble may be deferred with rationale).

Issue: #1164

---
_Generated by [Claude Code](https://claude.ai/code)_

## Summary by CodeRabbit

* **New Features**
  * Added strategy-backed proposers to the CLI, including `propose-strategy` and `replay strategy`.
  * Enabled `replay tournament` to run strategy-backed proposers alongside existing proposers.
  * Added `--price-precision` support for handling sub-cent symbols.
* **Bug Fixes**
  * Tightened validation for warm-up/min-history requirements and duplicate reruns.
  * Improved unattended Stage 1 scheduling gating and attested-equity behavior for predictable sizing.
* **Documentation**
  * Updated paper trading guidance with the alternate Stage 1 proposal path and scheduler prerequisites.
* **Tests**
  * Added unit coverage for strategy proposals, replays, tournaments, and precision edge cases.

### CODE REVIEW

File: src/gpt_trader/cli/commands/ideas.py [hunk 1]
```diff
@@ -1310,11 +1436,40 @@ def _resolve_replay_min_history(
...
 
+# The baseline strategy family detects MA crossovers over a fixed 3-bar
+# lookback inside ``decide``; its windows are not CLI-tunable because the
+# parity lane replays the live configuration as-is.
+_STRATEGY_CROSSOVER_LOOKBACK = 3
+
+
+def _default_strategy_replay_min_history() -> int:
+    config = BaseStrategyConfig()
+    return max(config.long_ma_period, config.rsi_period + 1) + _STRATEGY_CROSSOVER_LOOKBACK
```
[the remark this answers is not in the document]
Comment (Solders-Girdles) [reply src/gpt_trader/cli/commands/ideas.py]: Fixed in 9b9178a2. Agreed — the extra 3-bar lookback made the replay floor stricter than the live strategy's own warm-up gate, so 20–22-candle snapshots the live `decide()` would evaluate were silently skipped, which defeats the parity purpose of this lane. `_default_strategy_replay_min_history()` now returns exactly `max(long_ma_period, rsi_period + 1)` (= 20 with the live defaults) and the floor/default/error message all reference that gate; tournament floors follow the same helper. Tests updated to pin the 20-candle floor and the snapshot count over the full warm-up window.
